### DISCUSSION TITLE
How can I remove a repository with an exploit for my game

### DISCUSSION THREAD

**Discussion by alinnit** [community/community#67804] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Good day people,

I've discovered a repository with a LUA script that gives the players of my game the chance to edit the game files and remove the need for in game activity and payments in order to progress.

As you can understand this is very disruptive to my game and  of course to the multyplayer aspect of the game. Many of the players started complaining about cheaters.

Please let me know what are the steps I can take in order to remove the repository from GitHub.

Thank you!

**Comment by metux** [comment]
Did you ever consider fixing your broken code ?

**Comment by nitink3450** [comment, marked as answer]
If you've discovered a repository on GitHub that contains a LUA script that allows players to cheat in your game and disrupt the intended gameplay, you can take the following steps to request the removal of the repository:

1. **Contact the Repository Owner:**
   - Start by trying to contact the owner of the repository politely. They may not be aware of the negative impact their script is having on your game. Ask them to consider removing or disabling the script.

2. **File a DMCA Takedown Notice:**
   - If contacting the repository owner doesn't lead to a resolution, and if you believe that the script infringes on your intellectual property rights, you can file a Digital Millennium Copyright Act (DMCA) takedown notice with GitHub.
   - GitHub provides a DMCA agent contact and guidelines for filing a DMCA takedown notice. You can find this information on GitHub's DMCA policy page.

3. **Follow GitHub's Reporting Process:**
   - If the script doesn't directly infringe on your intellectual property rights but still violates GitHub's terms of service (e.g., by promoting cheating or harassment), you can report the repository to GitHub for violating their community guidelines. You can do this through the GitHub repository or user profile.

4. **Provide Documentation:**
   - When filing a DMCA takedown notice or reporting the repository to GitHub, make sure to provide any necessary documentation to support your case. This may include proof of ownership of the game and evidence of how the script negatively impacts gameplay.

5. **Be Patient and Persistent:**
   - After taking these steps, you may need to wait for GitHub to review your request and take appropriate action. Be patient, but if you believe your request is not being addressed, you can follow up with GitHub.

Remember that GitHub takes issues of copyright infringement and violations of their terms of service seriously. However, the process may take some time, and there is no guarantee of an immediate takedown. It's important to provide clear and compelling evidence to support your case.

Additionally, consider implementing measures within your game to prevent cheating and unauthorized script modifications in the future, as addressing individual repositories may not prevent new instances from appearing.

**Reply by alinnit** [reply]
Thank you for your detailed answer. Point 3 should be the way, as the legal department does not wish to get implicated with a DMCA claim.

**Reply by metux** [reply]
You should be aware that such a take down inevitably triggers the Streisand effect,
eg. by exploits going through various security mailing lists (fulldisclosure + friends),
appearing in dozens of forums, spread via twitter, telegram, etc, etc.

If this is a commercial game, the business impact could be very ugly.

The best approach IMHO is *politely* speaking w/ the exploit author and fixing the bug in your game.

**Reply by alinnit** [reply]
@metux Yep, not trying to cause any waves, just curious. Thanks for the message.
